src/offset_log.rs

```diff
@@ -158,69 +82,62 @@ impl<ByteType> OffsetLog<ByteType> {
             .expect("Unable to get metadata of file")
             .len();
 
-        let offset_codec = OffsetCodec::<ByteType>::with_length(file_length);
+        OffsetLog {
+            file,
+            end_of_file: file_length,
+            tmp_buffer: BytesMut::new(),
+            byte_type: PhantomData,
+        }
+    }
 
-        OffsetLog { offset_codec, file }
+    pub fn read(&self, offset: u64) -> Result<ReadResult, Error> {
+        read_forward::<ByteType, _>(offset, &self.file)
     }
+
     pub fn append_batch(&mut self, buffs: &[&[u8]]) -> Result<Vec<u64>, Error> {
-        let bytes = BytesMut::new();
+        let mut bytes = BytesMut::new();
         let mut offsets = Vec::<u64>::new();
 
-        let encoded = buffs.iter().try_fold(bytes, |mut acc, buff| {
-            let mut vec = Vec::new();
-
+        let new_end = buffs.iter().try_fold(self.end_of_file, |offset, buff| {
             //Maybe there's a more functional way of doing this. Kinda mixing functional and
             //imperative.
-            offsets.push(self.offset_codec.length);
-            vec.extend_from_slice(buff);
-            self.offset_codec.encode(vec, &mut acc).map(|_| acc)
+            offsets.push(offset);
+            encode::<ByteType>(offset, &buff, &mut bytes)
         })?;
 
-        self.file.seek(SeekFrom::End(0))?;
-        self.file.write(&encoded)?;
+        self.file.write_at(&bytes, self.end_of_file as usize)?;
+        self.end_of_file = new_end;
 
         Ok(offsets)
     }
+
+    pub fn iter(&self) -> OffsetLogIter<ByteType> {
+        // TODO: what are the chances that try_clone() will fail?
+        //  I'd rather not return a Result<> here.
+        OffsetLogIter::new(self.file.try_clone().unwrap())
+    }
 }
 
 impl<ByteType> FlumeLog for OffsetLog<ByteType> {
-    fn get(&mut self, seq_num: u64) -> Result<Vec<u8>, Error> {
-        let mut frame_bytes = vec![0; 4];
-
-        self.file.seek(SeekFrom::Start(seq_num as u64))?;
-        self.file.read(&mut frame_bytes)?;
-
-        let data_size = size_of_framing_bytes::<ByteType>()
-            + (&frame_bytes[0..4]).read_u32::<BigEndian>().unwrap() as usize;
-
-        let mut buf = Vec::with_capacity(data_size);
-        unsafe { buf.set_len(data_size) };
-
-        self.file.seek(SeekFrom::Start(seq_num as u64))?;
-        self.file.read(&mut buf)?;
-        self.offset_codec
-            .decode(&mut buf.into())?
-            .map(|val| val.data_buffer)
-            .ok_or(FlumeOffsetLogError::DecodeBufferSizeTooSmall {}.into())
+    fn get(&self, seq_num: u64) -> Result<Vec<u8>, Error> {
+        self.read(seq_num).map(|r| r.entry.data_buffer)
     }
 
     fn latest(&self) -> u64 {
-        self.offset_codec.last_valid_offset
+        unimplemented!()
```

Comment:
Is this a TODO or is it not needed or hard to do?

Reply:
Oh yeah, I'm not sure about that. Iirc, it used to be last-decoded-log-entry, so it was the offset of the most recent read, which doesn't seem very useful. We could probably use last-appended-entry?

Reply:
I guess I mean offset of last entry, which may not have been appended (if the file was just opened). Maybe I'll check the offset of the last entry on file open, then update it on every append, and make it available via latest()